yasnippet.el

```diff
@@ -3383,9 +3394,11 @@ If so cleans up the whole snippet up."
                 (t
                  nil)))))
     (unless (or (null snippets) snippets-left)
-      (if snippet-exit-transform
-          (yas--eval-for-effect snippet-exit-transform))
-      (yas--safely-run-hook snippet-exit-hook))))
+      (when snippet-exit-transform
+        (yas--eval-for-effect snippet-exit-transform))
+      (if (not (eq snippet-exit-hook-env yas-after-exit-snippet-hook))
```

Comment:
I believe this `snippet-exit-hook-env` is an unneeded complication.

> It's a way to check that yas-after-exit-hook was overridden in yas--letenv. I didn't know if the snippet had access to the outer scope.

Just keep the `snippet-exit-hook` variable, it gets set to the last snippet's `yas-after-exit-hook` value. Then at the end you can just `(let ((yas-after-exit-snippet-hook snippet-hook)) (yas--safe-run-hooks 'yas-after-exit-snippet-hook))`